### PR TITLE
fix: $routes->setDefaultMethod() does not work

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
- `$routes->setDefaultMethod()` does not work when you go to `/`

See https://forum.codeigniter.com/thread-81269.html

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by iRedds** [comment]
Wrong approach.
The constructor of the Router class already defines a controller and a default method.
And the Router::setDefaultController() method doesn't make sense.

**Comment by kenjis** [comment]
Removing a protected method is a breaking change.

I reverted ``Router::setDefaultController()`` and make it deprecated.

File: system/Router/Router.php [hunk 1]
```diff
@@ -646,8 +640,6 @@ protected function setRequest(array $segments = [])
     {
         // If we don't have any segments - try the default controller;
```
Comment (paulbalandan) [inline system/Router/Router.php:641]: This comment is no longer true

File: system/Router/Router.php [hunk 2]
```diff
@@ -462,15 +459,12 @@ public function autoRoute(string $uri)
...
 
+        // WARNING: Directories get shifted out of the segments array.
         $segments = $this->scanControllers($segments);
 
-        // If we don't have any segments left - try the default controller;
-        // WARNING: Directories get shifted out of the segments array.
-        if (empty($segments)) {
-            $this->setDefaultController();
-        }
+        // If we don't have any segments left - use the default controller;
```
Comment (paulbalandan) [inline system/Router/Router.php:465]: I think this comment should also be removed?

File: user_guide_src/source/changelogs/v4.2.0.rst [hunk 3]
```diff
@@ -42,6 +42,7 @@ Deprecations
 - ``CodeIgniter\Database\SQLSRV\Connection::getError()`` is deprecated. Use ``CodeIgniter\Database\SQLSRV\Connection::error()`` instead.
 - ``CodeIgniter\Debug\Exceptions::cleanPath()`` and ``CodeIgniter\Debug\Toolbar\Collectors\BaseCollector::cleanPath()`` are deprecated. Use the ``clean_path()`` function instead.
 - ``CodeIgniter\Log\Logger::cleanFilenames()`` and ``CodeIgniter\Test\TestLogger::cleanup()`` are both deprecated. Use the ``clean_path()`` function instead.
+- ``CodeIgniter\Router\setDefaultController()`` is deprecated.
```
Comment (paulbalandan) [inline user_guide_src/source/changelogs/v4.2.0.rst:45]:
```suggestion
- ``CodeIgniter\Router\Router::setDefaultController()`` is deprecated.
```

File: system/Router/Router.php (hunk 2 above)
Comment (kenjis) [reply system/Router/Router.php:465]: It is true. I don't think to be removed.

File: system/Router/Router.php (hunk 1 above)
Comment (kenjis) [reply system/Router/Router.php]: Indeed. "try" is strange.
Updated.

**Review by paulbalandan — APPROVED** [review]

**Pull request merged**